js/battledata.js

```diff
@@ -1024,6 +1025,18 @@ var Tools = {
 				}
 				template.formeid = formeid;
 			}
+			if (template.otherForms) {
+				var form = '', formid = '';
+				if (template.otherForms.indexOf(speciesid) >= 0) {
+					form = speciesid.slice(template.baseSpecies.length);
+					formid = '-' + form;
+					form = form[0].toUpperCase() + form.slice(1);
+				}
+				template.form = form;
```

Comment:
Yeah, I don't know if that's the best solution, but I think it's a better solution than what you're currently doing.